src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,131 @@
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { useTodos } from '../../context/TodosContext';
+import { TodoId } from '../../types/TodoId';
+import { useEffect, useRef, useState } from 'react';
+import { updateTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const editInputRef = useRef<HTMLInputElement>(null);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editValue, setEditValue] = useState(todo.title);
+
+  const {
+    currentId,
+    handleToggleTodoCheck,
+    handleDeletingTodo,
+    handleUpdateTodo,
+    handleSetError,
+    dispatch,
+  } = useTodos();
+
+  const isTemptTodo = (todoIdValue: TodoId) => todoIdValue === 0;
+
+  useEffect(() => {
+    if (isEditing && editInputRef.current) {
+      editInputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  const isEditingOrDeleting = async (event: React.FormEvent) => {
+    event.preventDefault();
+    if (todo.title === editValue.trim()) {
+      setIsEditing(false);
+
+      return;
+    }
+
+    if (editValue) {
+      dispatch({ type: 'todos/setCurrentId', payload: todo.id });
+      setIsEditing(true);
+      try {
+        await updateTodo(todo.id, { ...todo, title: editValue.trim() });
+
+        handleUpdateTodo(todo.id, { ...todo, title: editValue.trim() });
```

Comment:
It looks strange, `handleUpdateTodo` actually doesn't update todo? Why do you need them both